l10n_es_aeat_mod111/models/mod111.py

```diff
@@ -0,0 +1,274 @@
+        states={"calculated": [("readonly", False)]},
+        help="Capital gains derived from the forest exploitation of "
+        "residents in public forests - In kind - Base taxable value",
+    )
+    casilla_24 = fields.Float(
+        string="[24] Amount retained",
+        readonly=True,
+        states={"calculated": [("readonly", False)]},
+        help="Capital gains derived from the forest exploitation of "
+        "residents in public forests - In kind - Amount retained",
+    )
+    casilla_25 = fields.Integer(
+        string="[25] # Recipients",
+        readonly=True,
+        states={"calculated": [("readonly", False)]},
+        help="Consideration for the transfer of image rights: income "
+        "account provided in Article 92.8 of the Tax Law - "
+        "Monetary or in kind - Number of recipients",
+    )
+    casilla_26 = fields.Float(
+        string="[26] Taxable",
+        readonly=True,
+        states={"calculated": [("readonly", False)]},
+        help="Consideration for the transfer of image rights: income "
+        "account provided in Article 92.8 of the Tax Law - "
+        "Monetary or in kind - Base taxable value",
+    )
+    casilla_27 = fields.Float(
+        string="[27] Amount retained",
+        readonly=True,
+        states={"calculated": [("readonly", False)]},
+        help="Consideration for the transfer of image rights: income "
+        "account provided in Article 92.8 of the Tax Law - "
+        "Monetary or in kind - Amount retained",
+    )
+    casilla_28 = fields.Float(
+        string="[28] Amount of retentions",
+        compute_sudo=True,
+        readonly=True,
+        compute="_compute_casilla_28",
+        help="Amount of retentions: "
+        "([03] + [06] + [09] + [12] + [15] + [18] + [21] + [24] + [27])",
+    )
+    casilla_29 = fields.Float(
+        string="[29] Fees to compensate",
+        readonly=True,
+        states={"calculated": [("readonly", False)]},
+        help="Fee to compensate for prior results with same subject, "
+        "fiscal year and period (in which his statement was to return "
+        "and compensation back option was chosen).",
+    )
+    casilla_30 = fields.Float(
+        string="[30] Result",
+        readonly=True,
+        compute="_compute_casilla_30",
+        help="Result: ([28] - [29])",
+    )
+    tipo_declaracion = fields.Selection(
+        selection=[
+            ("I", "To enter"),
+            ("U", "Direct debit"),
+            ("G", "To enter on CCT"),
+            ("N", "To return"),
+        ],
+        string="Result type",
+        readonly=True,
+        default="I",
+        states={"draft": [("readonly", False)]},
+        required=True,
+    )
+    colegio_concertado = fields.Boolean(
+        string="College concerted",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=False,
+    )
+
+    @api.depends("tax_line_ids", "tax_line_ids.move_line_ids.partner_id")
```

Comment:
Hola @MiquelRForgeFlow  ,
hemos encontrado que existe un problema con los cálculos de las casillas al lanzar la acción de 
`def calculate(self)` , debido a un retraso de la escritura de la cache de Odoo en el sistema.
Una manera de arreglarlo sería indicando al principio de cada función de campos calculados el siguiente código:
`self.flush()
 self.invalidate_cache()`

pero obviamente esto no es mantenible... por lo que una sugerencia podría ser cambiar la función base de `def calculate(self)` en el módulo l10n_es_aeat para que realice esas dos funciones y al terminar el cálculo te grabe todo en memoria.
'Archivo: 10n_es_aeat_report.py' Linea: 433

¿Qué pensáis? @pedrobaeza @JordiBForgeFlow @acysos ......

Reply:
solucionado en el l10n_es_aeat